env_checker doesn't check if action space is dict

**Describe the bug**
An environment with a Dict Action Space, but a non-Dict non-Tuple Observation Space, will fail the `_check_nan(env)` [check](https://github.com/DLR-RM/stable-baselines3/blob/97b81f9e9ee2e5ba7eb37328bbd21f8eade44e72/stable_baselines3/common/env_checker.py#L238) because of `TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''`. 

I believe the fix to this is just adding a check that the action space is non-tuple, non-dict [here](https://github.com/DLR-RM/stable-baselines3/blob/97b81f9e9ee2e5ba7eb37328bbd21f8eade44e72/stable_baselines3/common/env_checker.py#L237).

**Full Error Output:**
```bash
tests/test_uwrt_arm_env.py:74 (TestClass.test_gym_api_compliance_for_dqn_wrapper_setup)
self = <test_uwrt_arm_env.TestClass object at 0x7f8a5bd2b5d0>

    def test_gym_api_compliance_for_dqn_wrapper_setup(self):
        env = FlattenObservation(MultiDiscreteToContinuousDictActionWrapper(
            gym.make('uwrt-arm-v0', key_position=self.KEY_POSITION, key_orientation=self.KEY_ORIENTATION,
                     max_steps=self.MAX_STEPS, enable_render=True)))

        # Implicitly closes the environment
>       env_checker.check_env(env=env, warn=True, skip_render_check=False)

test_uwrt_arm_env.py:83: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/wmmc88/anaconda3/envs/uwrt_arm_rl/lib/python3.7/site-packages/stable_baselines3/common/env_checker.py:238: in check_env
    _check_nan(env)
/home/wmmc88/anaconda3/envs/uwrt_arm_rl/lib/python3.7/site-packages/stable_baselines3/common/env_checker.py:74: in _check_nan
    _, _, _, _ = vec_env.step(action)
/home/wmmc88/anaconda3/envs/uwrt_arm_rl/lib/python3.7/site-packages/stable_baselines3/common/vec_env/base_vec_env.py:149: in step
    self.step_async(actions)
/home/wmmc88/anaconda3/envs/uwrt_arm_rl/lib/python3.7/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py:29: in step_async
    self._check_val(async_step=True, actions=actions)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <stable_baselines3.common.vec_env.vec_check_nan.VecCheckNan object at 0x7f8a5bce2510>
async_step = True
kwargs = {'actions': array([OrderedDict([('joint_velocity_commands', array([1, 1, 0, 1, 0]))])],
      dtype=object)}
found = [], name = 'actions'
val = array([OrderedDict([('joint_velocity_commands', array([1, 1, 0, 1, 0]))])],
      dtype=object)

    def _check_val(self, *, async_step, **kwargs):
        # if warn and warn once and have warned once: then stop checking
        if not self.raise_exception and self.warn_once and self._user_warned:
            return
    
        found = []
        for name, val in kwargs.items():
>           has_nan = np.any(np.isnan(val))
E           TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

/home/wmmc88/anaconda3/envs/uwrt_arm_rl/lib/python3.7/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py:58: TypeError
```

**System Info**
Describe the characteristic of your environment:
 * Describe how the library was installed (pip, docker, source, ...): pip
 * GPU models and configuration: 1080TI
 * Python version: Python 3.7.8
 * PyTorch version: 1.6.0
 * Gym version: 0.17.3
 * Versions of any other relevant libraries: N/A

--- a/docs/misc/changelog.rst
+++ b/docs/misc/changelog.rst
@@ -20,6 +20,7 @@ Bug Fixes:
 - Fix ignoring the exclude parameter when recording logs using json, csv or log as logging format (@SwamyDev)
 - Make ``make_vec_env`` support the ``env_kwargs`` argument when using an env ID str (@ManifoldFR)
 - Fix ``check_env`` not checking if the env has a Dict actionspace before calling ``_check_nan`` (@wmmc88)
+- Update the check for spaces unsupported by Stable Baselines 3 to include checks on the action space (@wmmc88)
 
 Deprecations:
 ^^^^^^^^^^^^^

--- a/docs/misc/changelog.rst
+++ b/docs/misc/changelog.rst
@@ -19,6 +19,7 @@ Bug Fixes:
 - Fix GAE computation for on-policy algorithms (off-by one for the last value) (thanks @Wovchena)
 - Fix ignoring the exclude parameter when recording logs using json, csv or log as logging format (@SwamyDev)
 - Make ``make_vec_env`` support the ``env_kwargs`` argument when using an env ID str (@ManifoldFR)
+- Fix ``check_env`` not checking if the env has a Dict actionspace before calling ``_check_nan`` (@wmmc88)
 
 Deprecations:
 ^^^^^^^^^^^^^
@@ -456,4 +457,4 @@ And all the contributors:
 @MarvineGothic @jdossgollin @SyllogismRXS @rusu24edward @jbulow @Antymon @seheevic @justinkterry @edbeeching
 @flodorner @KuKuXia @NeoExtended @PartiallyTyped @mmcenta @richardwu @kinalmehta @rolandgvc @tkelestemur @mloo3
 @tirafesi @blurLake @koulakis @joeljosephjin @shwang @rk37 @andyshih12 @RaphaelWag @xicocaio
-@diditforlulz273 @liorcohen5 @ManifoldFR @mloo3 @SwamyDev
+@diditforlulz273 @liorcohen5 @ManifoldFR @mloo3 @SwamyDev @wmmc88

--- a/stable_baselines3/common/env_checker.py
+++ b/stable_baselines3/common/env_checker.py
@@ -15,12 +15,14 @@ def _enforce_array_obs(observation_space: spaces.Space) -> bool:
     """
     return not isinstance(observation_space, (spaces.Dict, spaces.Tuple))
 
+
 def _enforce_array_act(action_space: spaces.Space) -> bool:
     """
     Whether to check that the returned action is a numpy array
     `Dict` and `Tuple` spaces are not guaranteed to contain only numpy arrays
     """
     return not isinstance(action_space, (spaces.Dict, spaces.Tuple))
+
 
 def _check_image_input(observation_space: spaces.Box) -> None:
     """

--- a/stable_baselines3/common/env_checker.py
+++ b/stable_baselines3/common/env_checker.py
@@ -8,20 +8,12 @@ from gym import spaces
 from stable_baselines3.common.vec_env import DummyVecEnv, VecCheckNan
 
 
-def _enforce_array_obs(observation_space: spaces.Space) -> bool:
+def _is_numpy_array_space(space: spaces.Space) -> bool:
     """
-    Whether to check that the returned observation is a numpy array
-    `Dict` and `Tuple` spaces are not guaranteed to contain only numpy arrays
+    Checks whether the space consists of a numpy array
+    `Dict` and `Tuple` spaces are not just a numpy array
     """
-    return not isinstance(observation_space, (spaces.Dict, spaces.Tuple))
-
-
-def _enforce_array_act(action_space: spaces.Space) -> bool:
-    """
-    Whether to check that the returned action is a numpy array
-    `Dict` and `Tuple` spaces are not guaranteed to contain only numpy arrays
-    """
-    return not isinstance(action_space, (spaces.Dict, spaces.Tuple))
+    return not isinstance(space, (spaces.Dict, spaces.Tuple))
 
 
 def _check_image_input(observation_space: spaces.Box) -> None:
@@ -53,7 +45,7 @@ def _check_image_input(observation_space: spaces.Box) -> None:
         )
 
 
-def _check_unsupported_obs_spaces(env: gym.Env, observation_space: spaces.Space) -> None:
+def _check_unsupported_spaces(env: gym.Env, observation_space: spaces.Space, action_space: spaces.Space) -> None:
     """Emit warnings when the observation space used is not supported by Stable-Baselines."""
 
     if isinstance(observation_space, spaces.Dict) and not isinstance(env, gym.GoalEnv):
@@ -71,6 +63,13 @@ def _check_unsupported_obs_spaces(env: gym.Env, observation_space: spaces.Space)
             "this is currently not supported by Stable Baselines "
             "(cf https://github.com/hill-a/stable-baselines/issues/133), "
             "you will need to flatten the observation and maybe use a custom policy. "
+        )
+
+    if not _is_numpy_array_space(action_space):
+        warnings.warn(
+            "The action space is not based off a numpy array. Typically this means it's either a Dict or Tuple space. "
+            "This type of action space is currently not supported by Stable Baselines 3. You should try to flatten the "
+            "action using a wrapper."
         )
 
 
@@ -95,7 +94,7 @@ def _check_obs(obs: Union[tuple, dict, np.ndarray, int], observation_space: spac
     # The check for a GoalEnv is done by the base class
     if isinstance(observation_space, spaces.Discrete):
         assert isinstance(obs, int), "The observation returned by `{}()` method must be an int".format(method_name)
-    elif _enforce_array_obs(observation_space):
+    elif _is_numpy_array_space(observation_space):
         assert isinstance(obs, np.ndarray), "The observation returned by `{}()` method must be a numpy array".format(
             method_name
         )
@@ -209,7 +208,7 @@ def check_env(env: gym.Env, warn: bool = True, skip_render_check: bool = True) -
     # Warn the user if needed.
     # A warning means that the environment may run but not work properly with Stable Baselines algorithms
     if warn:
-        _check_unsupported_obs_spaces(env, observation_space)
+        _check_unsupported_spaces(env, observation_space, action_space)
 
         # If image, check the low and high values, the type and the number of channels
         # and the shape (minimal value)
@@ -242,5 +241,5 @@ def check_env(env: gym.Env, warn: bool = True, skip_render_check: bool = True) -
         _check_render(env, warn=warn)
 
     # The check only works with numpy arrays
-    if _enforce_array_obs(observation_space) and _enforce_array_act(action_space):
+    if _is_numpy_array_space(observation_space) and _is_numpy_array_space(action_space):
         _check_nan(env)

--- a/stable_baselines3/common/env_checker.py
+++ b/stable_baselines3/common/env_checker.py
@@ -11,10 +11,16 @@ from stable_baselines3.common.vec_env import DummyVecEnv, VecCheckNan
 def _enforce_array_obs(observation_space: spaces.Space) -> bool:
     """
     Whether to check that the returned observation is a numpy array
-    it is not mandatory for `Dict` and `Tuple` spaces.
+    `Dict` and `Tuple` spaces are not guaranteed to contain only numpy arrays
     """
     return not isinstance(observation_space, (spaces.Dict, spaces.Tuple))
 
+def _enforce_array_act(action_space: spaces.Space) -> bool:
+    """
+    Whether to check that the returned action is a numpy array
+    `Dict` and `Tuple` spaces are not guaranteed to contain only numpy arrays
+    """
+    return not isinstance(action_space, (spaces.Dict, spaces.Tuple))
 
 def _check_image_input(observation_space: spaces.Box) -> None:
     """
@@ -234,5 +240,5 @@ def check_env(env: gym.Env, warn: bool = True, skip_render_check: bool = True) -
         _check_render(env, warn=warn)
 
     # The check only works with numpy arrays
-    if _enforce_array_obs(observation_space):
+    if _enforce_array_obs(observation_space) and _enforce_array_act(action_space):
         _check_nan(env)

--- a/stable_baselines3/common/env_checker.py
+++ b/stable_baselines3/common/env_checker.py
@@ -10,8 +10,8 @@ from stable_baselines3.common.vec_env import DummyVecEnv, VecCheckNan
 
 def _is_numpy_array_space(space: spaces.Space) -> bool:
     """
-    Checks whether the space consists of a numpy array
-    `Dict` and `Tuple` spaces are not just a numpy array
+    Returns False if provided space is not representable as a single numpy array
+    (e.g. Dict and Tuple spaces return False)
     """
     return not isinstance(space, (spaces.Dict, spaces.Tuple))
 
@@ -46,7 +46,7 @@ def _check_image_input(observation_space: spaces.Box) -> None:
 
 
 def _check_unsupported_spaces(env: gym.Env, observation_space: spaces.Space, action_space: spaces.Space) -> None:
-    """Emit warnings when the observation space used is not supported by Stable-Baselines."""
+    """Emit warnings when the observation space or action space used is not supported by Stable-Baselines."""
 
     if isinstance(observation_space, spaces.Dict) and not isinstance(env, gym.GoalEnv):
         warnings.warn(

--- a/tests/test_env_checker.py
+++ b/tests/test_env_checker.py
@@ -6,10 +6,7 @@ from stable_baselines3.common.env_checker import check_env
 
 
 class ActionDictTestEnv(gym.Env):
-    action_space = Dict({
-        "position": Discrete(1),
-        "velocity": Discrete(1)
-    })
+    action_space = Dict({"position": Discrete(1), "velocity": Discrete(1)})
     observation_space = Box(low=-1.0, high=2.0, shape=(3,), dtype=np.float32)
 
     def step(self, action):
@@ -22,7 +19,7 @@ class ActionDictTestEnv(gym.Env):
     def reset(self):
         return np.array([1.0, 1.5, 0.5])
 
-    def render(self, mode='human'):
+    def render(self, mode="human"):
         pass
 
 

--- a/tests/test_env_checker.py
+++ b/tests/test_env_checker.py
@@ -1,5 +1,6 @@
 import gym
 import numpy as np
+import pytest
 from gym.spaces import Box, Dict, Discrete
 
 from stable_baselines3.common.env_checker import check_env
@@ -25,4 +26,6 @@ class ActionDictTestEnv(gym.Env):
 
 def test_check_env_dict_action():
     test_env = ActionDictTestEnv()
-    check_env(env=test_env, warn=True)
+
+    with pytest.warns(Warning):
+        check_env(env=test_env, warn=True)

--- a/tests/test_env_checker.py
+++ b/tests/test_env_checker.py
@@ -1,0 +1,31 @@
+import gym
+import numpy as np
+from gym.spaces import Box, Dict, Discrete
+
+from stable_baselines3.common.env_checker import check_env
+
+
+class ActionDictTestEnv(gym.Env):
+    action_space = Dict({
+        "position": Discrete(1),
+        "velocity": Discrete(1)
+    })
+    observation_space = Box(low=-1.0, high=2.0, shape=(3,), dtype=np.float32)
+
+    def step(self, action):
+        observation = np.array([1.0, 1.5, 0.5])
+        reward = 1
+        done = True
+        info = {}
+        return observation, reward, done, info
+
+    def reset(self):
+        return np.array([1.0, 1.5, 0.5])
+
+    def render(self, mode='human'):
+        pass
+
+
+def test_check_env_dict_action():
+    test_env = ActionDictTestEnv()
+    check_env(env=test_env, warn=True)

